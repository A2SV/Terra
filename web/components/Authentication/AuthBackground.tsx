import React from "react";
import backgroundImage from "../../public/background.png";
import Image from "next/image";
import logo from "../../public/logo.png";

interface AuthBackgroundProps {
  description: string;
}

const AuthBackground: React.FC<AuthBackgroundProps> = ({ description }) => {
  return (
    <div className="flex items-center justify-center min-h-screen bg-white">
      <div className="flex md:flex-row bg-white shadow-lg min-h-screen w-full md:w-auto">
        <div className="hidden md:block relative w-1/2 overflow-hidden rounded-2xl p-8">
          <Image
            src={backgroundImage}
            alt="Background"
            className="object-cover rounded-2xl h-full w-full  ml-2"
          />
          <div className="absolute bottom-10 left-20 right-0 p-8 font-poppins">
            <h2 className="text-2xl font-semibold text-white">Terra is simply efficient</h2>
            <p className="mt-2 text-white">{description} today</p>
          </div>
        </div>
        <div className="w-full items-center md:w-1/2">
          <div className="flex justify-center flex-row md:justify-center">
            <Image
              src={logo}
              alt="Logo"
              height={150}
              width={150}
              className="object-cover mb-4 md:mb-0 md:mr-4"
            />
            <div className="mt-14 md:mt-14 ml-0 font-nunito ">
              <h2 className="font-semibold text-gray-700 text-2xl">Welcome to Terra</h2>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AuthBackground;
