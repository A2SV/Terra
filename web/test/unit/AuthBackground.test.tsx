import React from "react";
import { render, screen } from "@testing-library/react";
import AuthBackground from "@/components/Authentication/AuthBackground";

describe("AuthBackground", () => {
  it("renders without crashing", () => {
    render(<AuthBackground description="Test Description" />);
    expect(screen.getByText(/Welcome to Terra/i)).toBeInTheDocument();
  });

  it("displays the description prop correctly", () => {
    const testDescription = "Test Description";
    render(<AuthBackground description={testDescription} />);
    expect(screen.getByText(`${testDescription} today`)).toBeInTheDocument();
  });

  it("renders exactly two images", () => {
    render(<AuthBackground description="Test Description" />);
    const images = screen.getAllByRole("img");
    expect(images).toHaveLength(2);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<AuthBackground description="Snapshot Description" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
