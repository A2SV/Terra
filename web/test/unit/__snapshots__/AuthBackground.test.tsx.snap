// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBackground matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center min-h-screen bg-white"
  >
    <div
      class="flex md:flex-row bg-white shadow-lg min-h-screen w-full md:w-auto"
    >
      <div
        class="hidden md:block relative w-1/2 overflow-hidden rounded-2xl p-8"
      >
        <img
          alt="Background"
          class="object-cover rounded-2xl h-full w-full  ml-2"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="absolute bottom-10 left-20 right-0 p-8 font-poppins"
        >
          <h2
            class="text-2xl font-semibold text-white"
          >
            Terra is simply efficient
          </h2>
          <p
            class="mt-2 text-white"
          >
            Snapshot Description today
          </p>
        </div>
      </div>
      <div
        class="w-full items-center md:w-1/2"
      >
        <div
          class="flex justify-center flex-row md:justify-center"
        >
          <img
            alt="Logo"
            class="object-cover mb-4 md:mb-0 md:mr-4"
            data-nimg="1"
            decoding="async"
            height="150"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="150"
          />
          <div
            class="mt-14 md:mt-14 ml-0 font-nunito "
          >
            <h2
              class="font-semibold text-gray-700 text-2xl"
            >
              Welcome to Terra
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
